app/design/adminhtml/default/default/template/customer/address/attribute/js.phtml

```diff
@@ -0,0 +1,257 @@
+            if($('backend_type') && $('backend_type').options){
+                for(var i=0;i<$('backend_type').options.length;i++){
+                    if($('backend_type').options[i].value=='int') $('backend_type').selectedIndex = i;
+                }
+            }
+        }
+
+        switch ($('frontend_input').value) {
+            case 'textarea':
+                $('frontend_class').value = '';
+                $('frontend_class').disabled = true;
+                break;
+            case 'text':
+                if (!$('frontend_class').getAttribute('readonly')) {
+                    $('frontend_class').disabled = false;
+                }
+                break;
+            default:
+                $('frontend_class').value = '';
+                $('frontend_class').disabled = true;
+        }
+    }
+
+    function disableApplyToValue(value)
+    {
+        var applyToSelect = $('apply_to');
+        for (i=0;i<applyToSelect.options.length;i++) {
+            if (value == applyToSelect.options[i].value) {
+                applyToSelect.options[i].disabled = true;
+                applyToSelect.options[i].selected = false;
+            }
+        }
+    }
+
+    function switchDefaultValueField()
+    {
+        if (!$('frontend_input')) {
+            return;
+        }
+
+        var currentValue = $('frontend_input').value;
+
+        var defaultValueTextVisibility = false;
+        var defaultValueTextareaVisibility = false;
+        var defaultValueDateVisibility = false;
+        var defaultValueYesnoVisibility = false;
+        var scopeVisibility = true;
+
+        switch (currentValue) {
+            case 'select':
+                optionDefaultInputType = 'radio';
+                break;
+
+            case 'multiselect':
+                optionDefaultInputType = 'checkbox';
+                break;
+
+            case 'date':
+                defaultValueDateVisibility = true;
+                break;
+
+            case 'boolean':
+                defaultValueYesnoVisibility = true;
+                break;
+
+            case 'textarea':
+                defaultValueTextareaVisibility = true;
+                break;
+
+            case 'media_image':
+                defaultValueTextVisibility = false;
+                break;
+            case 'price':
+                scopeVisibility = false;
+            default:
+                defaultValueTextVisibility = true;
+                break;
+        }
+
+        switch (currentValue) {
+            case 'media_image':
+                $('front_fieldset').previous().hide();
+                $('front_fieldset').hide();
+
+                setRowVisibility('is_required', false);
+                setRowVisibility('is_unique', false);
+                setRowVisibility('frontend_class', false);
+                break;
+
+        <?php foreach (Mage::helper('catalog')->getAttributeHiddenFields() as $type=>$fields): ?>
```

Comment:
This method reads a value with key `attribute_type_hidden_fields` from registry, but from what I see it's not set anywhere in customer page.